### PR TITLE
 #846 and #742 FIX

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list

- FIX of #846. Update to Selenium 3.11
- #846 and #742 fixes. Migration to java.time.Duration
 
## Types of changes
- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
I am sorry for the long break. Was busy.

### CODE REVIEW

File: src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java [hunk 1]
```diff
@@ -59,11 +62,31 @@
      * @param by                a By locator strategy
      * @param shouldCache       is the flag that signalizes that elements which
      *                          are found once should be cached
-     * @param duration          is a POJO which contains timeout parameters for the element to be searched
+     * @param duration          is a POJO which contains timeout parameters for the element to be found
+     * @deprecated This constructor is going to be removed. Use {@link #AppiumElementLocator(SearchContext, By,
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java]: I 'd also add `@Deprecated` annotation to the constructor itself

File: src/main/java/io/appium/java_client/pagefactory/AppiumFieldDecorator.java [hunk 2]
```diff
@@ -62,16 +66,20 @@
     private static final List<Class<? extends WebElement>> availableElementClasses = ImmutableList.of(WebElement.class,
             RemoteWebElement.class, MobileElement.class, AndroidElement.class,
             IOSElement.class, WindowsElement.class);
+    public static Duration DEFAULT_WAITING_TIMEOUT = ofSeconds(1);
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/AppiumFieldDecorator.java]: final?

File: src/main/java/io/appium/java_client/pagefactory/WithTimeout.java [hunk 3]
```diff
@@ -38,7 +45,28 @@
...
+    ChronoUnit chronoUnit() default FOREVER;
+
+    class DurationBuilder {
+        static Duration build(WithTimeout withTimeout) {
+            //providing backward compatibility
+            if (!FOREVER.equals(withTimeout.chronoUnit())) {
+                return Duration.of(withTimeout.time(), withTimeout.chronoUnit());
+            }
+
+            return Duration.of(MILLISECONDS.convert(withTimeout.time(), withTimeout.unit()), MILLIS);
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/WithTimeout.java]: is FOREVER the same as milliseconds?

File: src/test/java/io/appium/java_client/android/BaseAndroidTest.java [hunk 4]
```diff
@@ -48,7 +48,7 @@
         File appDir = new File("src/test/java/io/appium/java_client");
         File app = new File(appDir, "ApiDemos-debug.apk");
         DesiredCapabilities capabilities = new DesiredCapabilities();
-        capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.ANDROID_UIAUTOMATOR2);
+        capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.APPIUM);
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/BaseAndroidTest.java]: why this is changed?

File: src/main/java/io/appium/java_client/pagefactory/WithTimeout.java (hunk 3 above)
Comment (TikhomirovSergey) [reply src/main/java/io/appium/java_client/pagefactory/WithTimeout.java]: I was forced to add defalt values (FOREVER and NANOSECONDS) for bacward compatibility (temporary). Comilation problems will occure otherewise. User will be forced to set both time units.

Comment (TikhomirovSergey) [reply src/main/java/io/appium/java_client/pagefactory/WithTimeout.java]: @mykola-mokhnach However
Can we consider FOREVER illegal value? I think it has some sense.

Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/pagefactory/WithTimeout.java]: I don't know, but it for sure does not help to read and understand the code.

**Pull request merged**